inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -147,12 +147,16 @@ def arxiv_plot_extract(obj, eng):
                 return
 
             if 'figures' in obj.data:
+                for figure in obj.data['figures']:
+                    del obj.files[figure['key']]
                 del obj.data['figures']
 
             lb = LiteratureBuilder(source='arxiv', record=obj.data)
-            for plot in plots:
+            for index, plot in enumerate(plots):
```

Comment:
This is still a bit confusing, but it works now that all the figures get deleted beforehand.